awswrangler/_utils.py

```diff
@@ -893,7 +894,11 @@ def split_pandas_frame(df: pd.DataFrame, splits: int) -> list[pd.DataFrame]:
 @engine.dispatch_on_engine
 def table_refs_to_df(tables: list[pa.Table], kwargs: dict[str, Any]) -> pd.DataFrame:
     """Build Pandas DataFrame from list of PyArrow tables."""
-    return _table_to_df(pa.concat_tables(tables, promote=True), kwargs=kwargs)
+    promote_kwargs: dict[str, bool | str] = {"promote": True}
+    if version.parse(pa.__version__) >= version.parse("14.0.0"):
+        promote_kwargs = {"promote_options": "default"}
```

Comment:
Can you clarify please - this was needed because schemas of the individual tables were different due to different index column types?

Reply:
This particular change wasn't needed for the fix to work. I just noticed a `FutureWarning` where the `promote` option had been replaced by `promote_options` in `pyarrow=14`.

The overall issue was that the index was dropped on write completely.

Reply:
Ok understood, thanks.